### DISCUSSION TITLE
The significance of position_embeddings

### DISCUSSION THREAD

**Discussion by Lnan1615** [Project-MONAI/MONAI#3260] [Q&A]
I cann't understand what the point of adding position_embeddings to the sequence is. For image blocks, when 2d or 3D images are transformed into 1d sequence or 1d sequence is transformed into 2D or 3D images, the operation is carried out in strict order, and the sequence sequence is not disrupted later, so it seems that position embedding of the sequence is no longer needed. In addition, I didn't understand the calculation method of position_embeddings. Could you give me a general explanation?

**Comment by wyli** [comment]
I assume you are talking about the vision transformers. The position embeddings are trainable position representations instead of hard-coded position indices. https://github.com/Project-MONAI/MONAI/blob/c9302e443b44c0f132beeb648b30313d94d7bccc/monai/networks/blocks/patchembedding.py#L99 cc @ahatamiz

**Reply by ahatamiz** [reply]
Hi @wyli and @Lnan1615 

In addition to the above, the position embedding is particularly used to retrain the spatial information with respect to each token in an input sequence. In a conventional CNN, the inductive image bias mitigates the need for such a mechanism, but in a ViT model ( or generally transformers), the locations of token need to be specified. This can be done via hard-coded (fixed) or trainable position embeddings. The ViT and UNETR both use the latter. 

Thanks
